runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -2874,6 +2916,40 @@ TR::Node *TR_VectorAPIExpansion::transformRORtoROL(TR_VectorAPIExpansion *opt, T
    return subNode;
    }
 
+bool
+TR_VectorAPIExpansion::getConvertSourceType(TR_VectorAPIExpansion *opt, TR::Node *node,
+                                            TR::DataType &sourceElementType, vec_sz_t &bitsLength)
+   {
+   TR::Compilation *comp = opt->comp();
+
+   // For convert, source vector type info is in children 2 and 3
+   TR::Node *sourceElementTypeNode = node->getChild(2);
+   sourceElementType = getDataTypeFromClassNode(comp, sourceElementTypeNode);
+
+   TR::Node *sourceNumLanesNode = node->getChild(3);
+
+   if (sourceNumLanesNode->getOpCode().isLoadConst())
+      {
+      int32_t elementSize = OMR::DataType::getSize(sourceElementType);
+      bitsLength = sourceNumLanesNode->get32bitIntegralValue()*8*elementSize;
+
+      if (supportedOnPlatform(comp, bitsLength) == TR::NoVectorLength)
+         {
+         traceMsg(comp, "Platform does not support conversion source length %d in node %p",
+                  bitsLength, node);
+         return false;
+         }
+      }
+
+   if (sourceElementType == TR::NoType || bitsLength == vec_len_default)
```

Comment:
When `sourceNumLanesNode` is not constant, the test `bitsLength == vec_len_default` here is relying on the caller to have initialized `bitsLength` to `vec_len_default` specifically. It looks like all callers do so, but it seems unnecessary to rely on that. What do you think of setting it to `vec_len_default` at the start of `getConvertSourceType()`?